Document Possible Issues with Apple M1 go Install

### Reproduction steps

### Environment

#### Hardware Overview:
  Model Name:	MacBook Air
  Model Identifier:	MacBookAir10,1
  Chip:	Apple M1
  Total Number of Cores:	8 (4 performance and 4 efficiency)
  Memory:	8 GB
  System Firmware Version:	7459.121.3
  OS Loader Version:	7459.121.3

### OS 
Monterey 12.4 
 
### Steps

1. Install okapi over brew
2. Run the test with : LD_LIBRARY_PATH=/opt/homebrew/lib go test ./... -run TestVaccineDemo


### Reproduction Code

Section of code. Note: I pulled this out from a larger piece of code and haven't tested this particular segment alone.

```

func makeTestEcosystem(tt *testing.T) (*services.Trinsic, *provider.CreateEcosystemResponse) {
	trinsic, err := services.NewTrinsic(services.WithTestEnv())
	assert.NoError(tt, err)

	provider := trinsic.Provider()
	assert.NoError(tt, err)

	eco, err := provider.CreateEcosystem(context.Background(), nil)
	assert.NoError(tt, err)

	return trinsic, eco
}

func createActors(t *testing.T, trinsic *services.Trinsic, ecoId string) map[string]string {
	actors := []string{allison, airline, clinic}
	ret := make(map[string]string)
	for _, a := range actors {
		actor, err := trinsic.Account().LoginAnonymous(context.Background(), ecoId)
		assert.NoError(t, err)
		ret[a] = actor
	}
	return ret
}

// Test here
func TestVaccineDemo(t *testing.T){
	var err error
	trinsic, eco := makeTestEcosystem(t)
	assert.NotNil(t, trinsic)

	actors := createActors(t, trinsic, eco.Ecosystem.Id)
	assert.Len(t, 3, actors)

	trinsic.SetAuthToken(actors[clinic])
	info, err := trinsic.Account().GetInfo(context.Background())
}
```

`LD_LIBRARY_PATH=/opt/homebrew/lib go test ./... -run TestVaccineDemo`

```
panic: unable to open a handle to the library [recovered]
        panic: unable to open a handle to the library

goroutine 8 [running]:
testing.tRunner.func1.2({0x4778200, 0xc0001acee0})
        /usr/local/go/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1399 +0x39f
panic({0x4778200, 0xc0001acee0})
        /usr/local/go/src/runtime/panic.go:884 +0x212
github.com/trinsic-id/okapi/go/okapi.getFunctionPointer({0x487cce6, 0xe})
        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/okapi/go@v1.6.0/okapi/native_call.go:81 +0x9d
github.com/trinsic-id/okapi/go/okapi.callOkapiNative({0x4a59fe0?, 0xc0000b1320?}, {0x4a5a000, 0xc000474820}, {0x487cce6, 0xe})
        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/okapi/go@v1.6.0/okapi/native_call.go:47 +0x152
github.com/trinsic-id/okapi/go/okapi.okapiMetadata.GetMetadata({})
        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/okapi/go@v1.6.0/okapi/metadata.go:20 +0x65
github.com/trinsic-id/sdk/go/services.(*serviceBase).BuildMetadata(0xc00010bf50, {0x4a5a360, 0xc0000b12c0})
        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/sdk/go@v1.8.0/services/service_base.go:120 +0xbf
github.com/trinsic-id/sdk/go/services.(*serviceBase).GetMetadataContext(0xc00009bba0?, {0x4a61e90, 0xc00003e0c8}, {0x4a5a360?, 0xc0000b12c0?})
        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/sdk/go@v1.8.0/services/service_base.go:104 +0x4d
github.com/trinsic-id/sdk/go/services.(*accountBase).GetInfo(0xc00007c300, {0x4a61e90, 0xc00003e0c8})        /Users/akmb2/go/pkg/mod/github.com/trinsic-id/sdk/go@v1.8.0/services/account_service.go:253 +0x62
github.com/benri-io/sdk/operators.TestVaccineDemo(0x0?)
        /Users/akmb2/workspace/github.com/benri-io/sdk/operators/trinsic_test.go:107 +0x12f
testing.tRunner(0xc00009bba0, 0x497f310)
        /usr/local/go/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1493 +0x35f
```

OK. I did a little more debugging. It's actually not CGO ( my bad ) but GOARCH not being set to arm64 that was causing the error. CGO also needs to be set, but my was actually already enabled by default. 

I think GOARCH defaults to `amd64`?

```
➜  operators git:(credential) ✗ go env GOOS GOARCH
darwin
amd64
```

But these are not explicitly set by me to be amd64:
```
➜  operators git:(credential) ✗ echo "os: $GOOS arch: $GOARCH"
os:  arch: 
``` 

FYI, when CGO is not set it gives a series of errors like this: 
```
/Users/akmb2/go/pkg/mod/github.com/trinsic-id/okapi/go@v1.6.1/okapi/didcomm.go:22:9: undefined: callOkapiNative
```

Whereas the GOARCH is incorrect, it gives the `panic: unable to open a handle to the library error.`

--- a/docs/go/index.md
+++ b/docs/go/index.md
@@ -3,6 +3,11 @@
 The Trinsic Go SDK makes it easy to interact with the Trinsic API from any Go application. The most recent version of the library can be found on ____. You can find the SDKs source on [Github](https://github.com/trinsic-id/sdk/tree/main/go).
 
 ## Installation and Configuration
+
+> If you're installing go on M1, the [Standard Go.dev Install Page](https://go.dev/doc/install) actually defaults to still showing you an amd64 install pkg even if you're on an M1.
+
+You have to go to the [other downloads](https://go.dev/dl/) page to get the `arm64` version for sure.
+
 * NOTE: Windows installation is possible, but requires jumping through a few hoops. More details to come later.
 * Make sure you have the okapi native binaries installed on your machine. Default installation instructions are in the [readme.md here](https://github.com/trinsic-id/okapi/)
 * You need to have [CGO configured](https://pkg.go.dev/cmd/cgo) for the okapi native binary link
@@ -22,3 +27,36 @@ Once the go package is installed and configured, you're ready to start building!
 
 [Start Walkthrough](../walkthroughs/vaccination.md){ .md-button .md-button--primary } [Explore API](../reference/index.md){ .md-button } [Go API Reference](../reference/index.md){ .md-button }
 
+## Diagnostics
+
+### M1 Architecture install issues
+
+```bash
+uname -m
+echo $GOARCH
+```
+
+The go architecture and apple processor architecture need to match for the dynamic okapi library loading to work.
+
+The long story short is though, when that happens, it actually sets all the environment variables also to AMD, including `GOHOSTARCH`, `GOARCH`, and `GOTOOLDIR`. This, then causes the bad env variable to impact the linker.
+
+If you receive an error that looks like the following on Apple M1 architecture, it is possible that you have the wrong architecture of golang installed/configured.
+```bash
+panic: unable to open a handle to the library [recovered]
+panic: unable to open a handle to the library
+```
+
+Try running a test code with the following env vars set:
+```bash
+GOARCH=arm64  go test ./...
+```
+
+If `CGO` is not enabled, you will receive an error like the following:
+```bash
+go/pkg/mod/github.com/trinsic-id/okapi/go@v1.6.1/okapi/didcomm.go:22:9: undefined: callOkapiNative
+```
+
+Try running with the following env var set:
+```bash
+CGO_ENABLED=1  go test ./...
+```

